aws/resource_aws_lb_target_group.go

```diff
@@ -495,9 +496,29 @@ func resourceAwsLbTargetGroupUpdate(d *schema.ResourceData, meta interface{}) er
 func resourceAwsLbTargetGroupDelete(d *schema.ResourceData, meta interface{}) error {
 	elbconn := meta.(*AWSClient).elbv2conn
 
-	_, err := elbconn.DeleteTargetGroup(&elbv2.DeleteTargetGroupInput{
+	input := &elbv2.DeleteTargetGroupInput{
 		TargetGroupArn: aws.String(d.Id()),
+	}
+
+	log.Printf("[DEBUG] Deleting Target Group (%s): %s", d.Id(), input)
+	err := resource.Retry(2*time.Minute, func() *resource.RetryError {
```

Comment:
We've been updating calls to `resource.Retry` to use `resource.StateChangeConf` and `WaitForState()`, which is a more robust way of waiting and handles a lot of boilerplate code.

Take a look at `resourceAwsRDSClusterParameterGroupDelete()` for an example of how this can be used.

Reply:
This would require a big refactor in how the lb_target_group CRUD functions are implemented and I think it's outside this PR scope. Do you think this is necessary or should I work on this in a separate PR? 😃

Reply:
Good point! A separate PR is fine